packages/shared/src/store/postActions.ts

```diff
@@ -414,6 +440,54 @@ export async function addPostReaction(
   }
 }
 
+/**
+ * Verifies whether a post was actually delivered to the server.
+ * This is used for posts that are marked as 'needs_verification' due to connection issues.
+ */
+export async function verifyPostDelivery(post: db.Post): Promise<boolean> {
+  logger.crumb('verifying post delivery', {
+    postId: post.id,
+    channelId: post.channelId,
+  });
+
+  if (post.deliveryStatus !== 'needs_verification') {
+    logger.crumb('post does not need verification', {
+      status: post.deliveryStatus,
+    });
+    return false;
+  }
+
+  try {
+    // Attempt to fetch the post from the server
+    await api.getPostWithReplies({
```

Comment:
will the `post.id` we have cached in the DB match what we need for this scry? 

If it got marked `needs_verification` won't that mean we haven't received the backend update with it's true canonical ID?

Reply:
Good point, this won't work as-is. We'd need a way to query for the post based on author and sent time.

Reply:
nice, change looks good!